xy_grid: Find hypernetworks by short name

Milly opened the pull request:
User can specify hypernetworks in xy-plot by short matched name instead of full name.
This is similar functionality to checkpoint in xy-plot.

AUTOMATIC1111 commented:
this can be merged in but it needs to be updated to account latest changes I think

Milly committed:
xy_grid: Find hypernetwork by closest name

Milly committed:
xy_grid: Confirm that hypernetwork options are valid before starting

Milly committed:
xy_grid: Refactor confirm functions

Milly commented:
Rebased and fixed.

### Additional feature
- Confirm that hypernetwork options are valid before starting. (63d1682)
- Refactored and generalized confirm functions. (170e345)
  - Added confirm field to `AxisOption` and `AxisOptionImg2Img`.
  - If you don't want this refactoring, skip this commit.
